mlrun/runtimes/nuclio/application/application.py

```diff
@@ -469,14 +489,22 @@ def create_api_gateway(
         api_gateway_scheme = db.store_api_gateway(
             api_gateway=api_gateway.to_scheme(), project=self.metadata.project
         )
-        if not self.status.api_gateway_name:
-            self.status.api_gateway_name = api_gateway_scheme.metadata.name
-        self.status.api_gateway = APIGateway.from_scheme(api_gateway_scheme)
-        self.status.api_gateway.wait_for_readiness()
-        self.url = self.status.api_gateway.invoke_url
 
-        logger.info("Successfully created API gateway", url=self.url)
-        return self.url
+        if set_as_default:
+            self.status.api_gateway_name = api_gateway_scheme.metadata.name
+            self.status.api_gateway = APIGateway.from_scheme(api_gateway_scheme)
+            self.status.api_gateway.wait_for_readiness()
+            self.url = self.status.api_gateway.invoke_url
+            url = self.url
```

Comment:
Maybe also delete the old default if one exists here?

Reply:
I believe in our discussion this was mentioned to happen automatically since you're using the same name for the GW, so nuclio will replace.

Reply:
Not always anymore, as this code allows you to create a GW with a different name and set it as the default, overriding anything that was there before